File: 0347-top-k-frequent-elements/0347-top-k-frequent-elements.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        priority_queue<pair<int,int>>pq;
        map<int,int>m;
        for(auto& n : nums) m[n]++;
        for(auto& e : m) {
            pq.push({e.second,e.first});
        }
        vector<int>v;
        for(int i=k; i; i--) {
            v.push_back(pq.top().second);
            pq.pop();
        }
        return v;
    }
};

File: 0347-top-k-frequent-elements/0347-top-k-frequent-elements.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        
        d = defaultdict(int)
        
        for n in nums:
            d[n] += 1
            
        s = sorted(d.items(), key=lambda item: item[1])
        
        return [item[0] for item in s[-k:]]